EventListener/CommentFormFieldsSubscriber.php

```diff
@@ -57,112 +79,29 @@ public function onBuildField(FieldEvent $event)
-
-                $getter = Inflector::camelize('get_'.$fieldName);
-                $setter = Inflector::camelize('set_'.$fieldName);
-
-                $body = $content->$getter();
-
-                if ($commentContent = $repository->findOneBy(['content.$id' => $content->getId(), 'field' => $fieldName])) {
-                    $commentContent->setBody($body);
-                    $this->documentManager->flush();
-                } else {
-                    $commentContent = new CommentContent();
-                    $commentContent->setContent($content);
-                    $commentContent->setField($fieldName);
-                    $commentContent->setBody($body);
-                    $this->documentManager->persist($commentContent);
-                    $this->documentManager->flush();
-                }
-
-                $stripComments = $this->stripComments($body);
-                $content->$setter($stripComments);
-            }
-        }
-    }
-
-    /**
-     * @param $body
-     * @return string
-     */
-    private function stripComments($body)
-    {
-        $domDocument = new \DOMDocument();
-        $domDocument->loadHTML($body);
-
-        $xpath = new \DOMXPath($domDocument);
-        $nodeList = $xpath->query("//span[contains(@class, 'comment-added')]");
-
-        /** @var \DOMElement $item */
-        foreach ($nodeList as $item) {
-            $itemHtml = $this->getInnerHtml($item);
+        $options = $field->getOptions();
 
-            $newNode = $domDocument->createTextNode($itemHtml);
-
-            $item->parentNode->replaceChild($newNode, $item);
+        $repository = $this->documentManager->getRepository('IntegratedCommentBundle:Comment');
+        //todo one query per request
+        if ($comment = $repository->findBy(['content.$id' => $content->getId(), 'field' => $field->getName()], ['date' => 'asc'])) {
+            $comment = $comment[0];
+            $options['attr'] = array('data-comment-id' => $comment->getId());
+            $field->setOptions($options);
         }
 
-        $html = $domDocument->saveHTML($xpath->query('//body')->item(0));
-
-        return preg_replace("/<\/?body>/i", "", $html);
+        $this->stylesheets->add('bundles/integratedcomment/css/comments.css');
+        $this->javascripts->add('bundles/integratedcomment/js/comments.js');
     }
 
     /**
-     * @param \DOMElement $domElement
-     * @return string
+     * @param FieldEvent $event
      */
-    private function getInnerHtml(\DOMElement $domElement)
+    public function postBuildField(BuilderEvent $event)
     {
-        $innerHTML = "";
+        $field = $event->getBuilder()->get($event->getField());
 
-        /** @var \DOMNode $child */
-        foreach ($domElement->childNodes as $child) {
-            $innerHTML .= $domElement->ownerDocument->saveHTML($child);
+        if ($field->getType()->getName() == 'integrated_editor') {
```

Comment:
A code standard in Symfony2 is to use Yoda statements. (see http://symfony.com/doc/current/contributing/code/standards.html#structure)